### PR TITLE
Pay 4982 refunds actions by reviewer

### PR DESCRIPTION
**Pull request by Jennings-HMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/refunds/config/RestTemplateConfiguration.java [hunk 1]
```diff
@@ -17,10 +17,14 @@ public CloseableHttpClient paymentsHttpClient() {
             .build();
     }
 
-    @Bean("restTemplateIdam")
+    @Bean ("restTemplateIdam")
     public RestTemplate restTemplateIdam() {
-        return new RestTemplate(new HttpComponentsClientHttpRequestFactory());
+        return  new RestTemplate(new HttpComponentsClientHttpRequestFactory());
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/config/RestTemplateConfiguration.java:22]: remove extra space

File: src/main/java/uk/gov/hmcts/reform/refunds/dtos/requests/RefundLiberataFee.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+@JsonInclude(NON_NULL)
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder(builderMethodName = "refundLiberataFeeWith")
+public class RefundLiberataFee {
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/dtos/requests/RefundLiberataFee.java:21]: Don't name a class after third party company

File: src/main/java/uk/gov/hmcts/reform/refunds/dtos/requests/RefundLiberataRequest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+@JsonInclude(NON_NULL)
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder(builderMethodName = "refundLiberataRequestWith")
+public class RefundLiberataRequest {
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/dtos/requests/RefundLiberataRequest.java:23]: Don't name a class after third party company name

File: src/main/java/uk/gov/hmcts/reform/refunds/dtos/requests/PaymentFeeDetails.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.List;
+
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+@Builder(builderMethodName = "paymentFeeDetailsWith")
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+public class PaymentFeeDetails {
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/dtos/requests/PaymentFeeDetails.java:19]: Follow the same pattern for naming classes

File: src/main/java/uk/gov/hmcts/reform/refunds/dtos/requests/PaymentFeeDetails.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    private String reference;
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/dtos/requests/PaymentFeeDetails.java:21]: call it paymentReference

File: src/main/java/uk/gov/hmcts/reform/refunds/dtos/responses/FeeDto.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+@JsonInclude(NON_NULL)
+@Builder(builderMethodName = "feeDtoWith")
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+public class FeeDto {
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/dtos/responses/FeeDto.java:26]: You don;t need all these fields. Only use the ones that payment actually returns

File: src/main/java/uk/gov/hmcts/reform/refunds/dtos/responses/PaymentAllocationDto.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+@JsonInclude(NON_NULL)
+@Builder(builderMethodName = "paymentAllocationDtoWith")
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+public class PaymentAllocationDto {
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/dtos/responses/PaymentAllocationDto.java:23]: Use same naming pattern

File: src/main/java/uk/gov/hmcts/reform/refunds/dtos/responses/PaymentDto.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+@JsonNaming(SnakeCaseStrategy.class)
+@JsonInclude(NON_NULL)
+@Builder(builderMethodName = "payment2DtoWith")
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+public class PaymentDto {
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/dtos/responses/PaymentDto.java:27]: Don't need all these fields

File: src/main/java/uk/gov/hmcts/reform/refunds/services/LiberataService.java [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+public class LiberataService {
+
+    @Value("${liberata.api.url}")
+    private String liberataApiUrl;
+
+    @Qualifier("restTemplatePayment")
+    @Autowired()
+    private RestTemplate restTemplatePayment;
+
+    private static final String LIBERATA_ENDPOINT = "/api/v3/refund";
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/LiberataService.java:36]: Move to configuration

File: src/main/java/uk/gov/hmcts/reform/refunds/services/LiberataService.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+    /**
+     *
+     * @param paymentDto
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/LiberataService.java:40]: remove Java doc

File: src/main/java/uk/gov/hmcts/reform/refunds/services/LiberataService.java [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+            .build();
+    }
+
+
+    private List<RefundLiberataFee> getRefundRequestFees(List<FeeDto> fees) {
+        return fees.stream().map(feeDto -> feeDtoMapToRefundLiberatFee(feeDto))
+            .collect(Collectors.toList());
+    }
+
+    private RefundLiberataFee feeDtoMapToRefundLiberatFee(FeeDto feeDto){
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/LiberataService.java:67]: Move to mapping class

File: src/main/java/uk/gov/hmcts/reform/refunds/services/RefundReviewServiceImpl.java [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+import uk.gov.hmcts.reform.refunds.state.RefundEvent;
+import uk.gov.hmcts.reform.refunds.state.RefundState;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.refunds.state.RefundState.SUBMITTED;
+
+@Service
+public class RefundReviewServiceImpl implements  RefundReviewService{
```
Comment (YasharRahvar) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/RefundReviewServiceImpl.java:37]: remove extra space

File: src/main/java/uk/gov/hmcts/reform/refunds/config/RestTemplateConfiguration.java (hunk 1 above)
Comment (Jennings-HMCTS) [reply src/main/java/uk/gov/hmcts/reform/refunds/config/RestTemplateConfiguration.java:22]: removed

**Pull request merged**